packages/cf-component-icon/src/Icon.js

```diff
@@ -1,33 +1,245 @@
 import React from 'react';
 import PropTypes from 'prop-types';
 
-class Icon extends React.Component {
-  render() {
-    let className = `cf-icon cf-icon--${this.props.type}`;
+import { createComponent } from '../../cf-style-container';
+
+const Wrapper = createComponent(({
+  color = '#333333',
+  fontSize = 'initial',
+  border = false,
+  spin = false,
+  type,
+  theme
+}) => {
+  const additionalRules = {};
+  let fontPoint = '';
+
+  switch (type) {
```

Comment:
Or maybe it's not needed since the `type` proptype is set to being a `required` prop.